front/src/api/books.js

```diff
@@ -1,8 +1,11 @@
+import axios from "axios";
+
 import { config } from "../config";
 
 export const books = {
-  fetchBooks: async (pageNumber) => {
-    const res = await fetch(`${config.url}/book?page=${pageNumber}`);
-    return await res.json();
+  fetchBooks: (pageNumber, dispatch, fetchBooksDataDataSucceeded) => {
```

Comment:
nie nie, tu przekazujesz tylko pageNumber :D a zwracasz response.data
potem w thunku odpalasz dispatch(fetchBooksDataDataSucceeded(data))

Tak jest nieładnie, bo przekazujesz funkcje do funkcji i musisz wracac zeby sprawdzic co ona robiła. Jezeli tu zwrocisz tylko response.data, to jest bardziej przejrzyscie.

- pobieram dane, przypisuje je do jakiejs zmiennej (const data = api.fetchBooks(pageNumber)
- obrabiam dane (dispatch(fetchBooksDataDataSucceeded(data))

Reply:
Done.